instructor/client_mistral.py

```diff
@@ -0,0 +1,64 @@
+import mistralai.client
+import mistralai.async_client as mistralaiasynccli
+import instructor
+from typing import Type, ParamSpec
```

Comment:
please merge in main and use the proper __future__ imports to make sure this is compatible with 3.9